scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -600,13 +603,31 @@ public static SelectionScript replaceBindingsInsideScript(SelectionScript script
      * Load and initialize the task to be started
      *
      * @param task the task to be initialized
+     * @param etd
      */
-    protected void loadAndInit(InternalTask task) {
+    protected void loadAndInit(InternalTask task, Iterator<EligibleTaskDescriptor> etd) {
         if ((task.getExecutableContainer() == null) ||
             ((ScriptExecutableContainer) task.getExecutableContainer()).getScript() == null) {
-            tlogger.debug(task.getId(), "initializing the executable container");
-            ExecutableContainer container = getDBManager().loadExecutableContainer(task);
-            task.setExecutableContainer(container);
+            try {
+                tlogger.debug(task.getId(), "initializing the executable container");
+                ExecutableContainer container = getDBManager().loadExecutableContainer(task);
+                task.setExecutableContainer(container);
+            } catch (Exception e) {
+                handleLoadTaskError(task, etd, e);
+            }
+        }
+    }
+
+    private void handleLoadTaskError(InternalTask task, Iterator<EligibleTaskDescriptor> etd, Exception e) {
+        String message = "Error when loading task " + task.toString() +
+                         " definition from the database. See server logs for more details. Cancelling task " +
+                         task.getId();
+        logger.error(message, e);
+        etd.remove();
```

Comment:
instead of removing task here (where we dont have original `taskRetrievedFromPolicy.iterator()`) maybe we can throw some exception, and catch where we have `taskRetrievedFromPolicy.iterator()`

Reply:
Ok I will have a look if we can catch the exception at a upper level